tests/check_consistent.py

```diff
@@ -8,57 +8,48 @@
 import os
 import re
 import sys
+from pathlib import Path
 
 import tomli
 from packaging.requirements import Requirement
 from packaging.version import Version
 
 metadata_keys = {"version", "requires", "extra_description", "obsolete_since", "no_longer_updated", "tool"}
 tool_keys = {"stubtest": {"skip", "apt_dependencies", "extras", "ignore_missing_stub"}}
-allowed_files = {"README.md"}
 
 
-def assert_stubs_only(directory: str) -> None:
+def assert_stubs_only(directory: Path, allowed: set[str]) -> None:
     """Check that given directory contains only valid stub files."""
-    top = directory.split(os.sep)[-1]
-    assert top.isidentifier(), f"Bad directory name: {top}"
-    for _, dirs, files in os.walk(directory):
-        for file in files:
-            if file in allowed_files:
-                continue
-            name, ext = os.path.splitext(file)
-            assert name.isidentifier(), f"Files must be valid modules, got: {name}"
-            assert ext == ".pyi", f"Only stub flies allowed. Got: {file} in {directory}"
-        for subdir in dirs:
-            assert subdir.isidentifier(), f"Directories must be valid packages, got: {subdir}"
+    allowed_paths = {Path(f) for f in allowed}
+    contents = list(directory.iterdir())
+    while contents:
```

Comment:
It would be simpler to for loop over `directory.rglob("*")`.

Reply:
That doesn't quite work, because currently if a subdirectory (e.g. `@tests`) is allowed, we won't check its contents

Reply:
We could do `if entry in allowed_paths or any(parent in allowed_paths for parent in entry.parents)`. A bit of a mess and recurses unnecessarily, but maybe more explicit. I'll let you decide :)

Reply:
Lol, you need to do the whole `relative_to()` thing for that check. It would become a bit complicated, maybe not worth the effort :)